specs/core/1_custody-game.md

```diff
@@ -263,45 +274,99 @@ class BeaconBlockBody(Container):
 
 ```python
 def ceillog2(x: uint64) -> int:
-    return x.bit_length()
+    return (x - 1).bit_length()
+```
+
+### `is_valid_merkle_branch_with_mixin`
+
+```python
+def is_valid_merkle_branch_with_mixin(leaf: Hash, 
```

Comment:
I noticed we need this function to check the Merkle proofs for Bitlists (and other variable types). Wasn't sure whether to put this as a modification in phase 0 (I think that would also be valid as it is such a small change and won't affect the phase 0 logic).

Reply:
I think we were thinking about modifying the proof of custody data to be a vector rather than a list, so the mixin is unnecessary. CC @JustinDrake. Really nailing down the shard chain details is a dependency of this.

Reply:
OK, so I will leave this as is for now and then if we are not going to need any length mixins, we can remove it.